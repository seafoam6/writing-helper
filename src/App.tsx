import React, { useState } from 'react';
import {Button} from 'grommet';
import styled from 'styled-components'

export const arrOfNumbers = (count: number): number[] => {
  let i = 0;
  const thing = [];
  while (i < count){
    i++;
    thing.push(i)
  }
  return thing;
}

export const setOfThings = (count: number, thingToDo: string): string[] => {
  return arrOfNumbers(count).map(i => `${thingToDo} ${i}`)
}

export const stuffToDo = [
  "read script",
 ...setOfThings(4, 'read outline'), 
 ...setOfThings(4, 'write part of'), 
 "work on character outline"
]

export const getRandomThing = () => {
  return stuffToDo[Math.floor(Math.random() * stuffToDo.length)];
}

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height:100vh
`

const App: React.FC = () => {
  const defaultText = 'PUSH ME'
   const [buttonText, setButtonText] = useState(defaultText);

  const handleClick = () => {
    const thingToDo =  getRandomThing();

    setButtonText(thingToDo);
  }

  return (
    <Wrapper>
      <Button onClick={handleClick} label={buttonText} />
      <Button onClick={()=> setButtonText(defaultText)} label='Reset' />
    </Wrapper>
  );
}

export default App;
