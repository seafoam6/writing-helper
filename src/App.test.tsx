import React from 'react';
import ReactDOM from 'react-dom';
import App, {arrOfNumbers, setOfThings, stuffToDo, getRandomThing} from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('arrOfNumbers', () => {
  it('returns an array of numbers', () => {
    expect(arrOfNumbers(4)).toEqual([1,2,3,4])
  })
})

describe('setOfThings', () => {
  it('returns an array of things', () => {
    expect(setOfThings(4, "DA DA DA")).toMatchSnapshot()
  })
})

describe('stuffToDo', () => {
  it('returns an array of things', () => {
    expect(stuffToDo).toMatchSnapshot()
  })
})

describe('getRandomThing', () => {
  it('returns an array of things', () => {
    const thing = getRandomThing();
    expect(stuffToDo.indexOf(thing)).not.toEqual(-1)
  })
})