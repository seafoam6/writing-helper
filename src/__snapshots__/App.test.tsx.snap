// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setOfThings returns an array of things 1`] = `
Array [
  "DA DA DA 1",
  "DA DA DA 2",
  "DA DA DA 3",
  "DA DA DA 4",
]
`;

exports[`stuffToDo returns an array of things 1`] = `
Array [
  "read script",
  "read outline 1",
  "read outline 2",
  "read outline 3",
  "read outline 4",
  "write part of 1",
  "write part of 2",
  "write part of 3",
  "write part of 4",
  "work on character outline",
]
`;
